test/linkify-matrix-test.ts

```diff
@@ -286,13 +311,21 @@ describe("linkify-matrix", () => {
             const found = linkify.find(test);
             expect(found).toEqual([
                 {
-                    href: char + "foo:bar.com-baz.com",
```

Comment:
The test title says "does _not_ parse" but then the test asserted that the full string was accepted. This also doesn't seem to be a valid room identifier so I suspect this was a mistake?

Reply:
Why would `#foo:bar.com-baz.com` not be a valid room identifier? `com-baz.com` looks like a valid domain, `bar` being a valid subdomain. Dashed in domains are valid, looks like this test being updated asserts they are not. https://www.fasthosts.co.uk/blog/hyphens-in-domain-names/

Reply:
Hm, yeah I think you're right. Hyphens are also allowed according to https://spec.matrix.org/v1.7/appendices/#server-name. But then the name of the test is sort of misleading?

It's also interesting that linkify's `domain` group doesn't appear to include hyphens.

Reply:
> But then the name of the test is sort of misleading?

Agreed

> It's also interesting that linkify's domain group doesn't appear to include hyphens.

Quite

Reply:
Have re-introduced the additional domainpart tokens to fix this